atom/browser/media/media_capture_devices_dispatcher.cc

```diff
@@ -49,75 +42,75 @@ MediaCaptureDevicesDispatcher::MediaCaptureDevicesDispatcher()
 
 MediaCaptureDevicesDispatcher::~MediaCaptureDevicesDispatcher() {}
 
-const MediaStreamDevices&
+const blink::MediaStreamDevices&
 MediaCaptureDevicesDispatcher::GetAudioCaptureDevices() {
   DCHECK(BrowserThread::CurrentlyOn(BrowserThread::UI));
   if (is_device_enumeration_disabled_)
-    return EmptyDevices();
+    return test_audio_devices_;
```

Comment:
What's the intent behind returning an internal list instead of making a new empty one? This seems like a potential issue if this somehow became non-empty

Reply:
This is already behind a noop code path until `DisableDeviceEnumerationForTesting` is called. I am inclined to remove this condition all together, the goal of rewriting this was to see if the upstream `MediaCaptureDevicesDispatcher` class can be reused with less patch.